src/App.tsx

```diff
@@ -1,24 +1,203 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { FilterTodos } from './types/FilterTodos';
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 6550;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Error.NONE);
+  const [filterBy, setFilterBy] = useState(FilterTodos.ALL);
+  const [activeInput, setActiveInput] = useState(true);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [idUpdating, setIdUpdating] = useState([0]);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        setErrorMessage(Error.NONE);
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(Error.LOAD);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const countTodos = todos.length;
+
+  const activeTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    switch (filterBy) {
+      case FilterTodos.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterTodos.COMPLTED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterBy]);
+
+  const addTodo = async (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(Error.TITLE);
+
+      return;
+    }
+
+    try {
+      setErrorMessage(Error.NONE);
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setActiveInput(false);
+      setTempTodo({ ...newTodo, id: 0 });
+      const createdTodo = await createTodo(USER_ID, newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setErrorMessage(Error.ADD);
+    } finally {
+      setTempTodo(null);
+      setActiveInput(true);
+    }
+  };
+
+  const removeTodo = async (id: number) => {
+    try {
+      setErrorMessage(Error.NONE);
+      setIdUpdating(prevId => [...prevId, id]);
+
+      await deleteTodo(USER_ID, id);
+
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch (error) {
+      setErrorMessage(Error.DELETE);
+    } finally {
+      setIdUpdating([0]);
+    }
+  };
+
+  const removeCompletedTodos = () => {
+    todos.forEach(async (todo) => {
```

Comment:
try to avoid using some `async` callbacks for array methods
read this [link](https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971)

Reply:
not fixed
![image](https://user-images.githubusercontent.com/98764641/227867659-6d233c5b-634f-49ae-a285-b7b310ff3ce8.png)